File: java/spring-boot/webapp.g8/src/main/g8/app/src/main/java/$package$/api/HelloController.java
package $package$.api;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

/** The API lives under {@code /api}: the one prefix the SPA fallback never answers for. */
@RestController
@RequestMapping("/api")
public class HelloController {

    public record Greeting(String message) {}

    @GetMapping("/hello")
    public Greeting hello(@RequestParam(defaultValue = "world") String name) {
        return new Greeting("Hello, " + name + "!");
    }
}

File: kotlin/spring-boot/webapp.g8/src/main/g8/app/src/main/kotlin/$package$/api/HelloController.kt
package $package$.api

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

/** The API lives under `/api`: the one prefix the SPA fallback never answers for. */
@RestController
@RequestMapping("/api")
class HelloController {

    data class Greeting(val message: String)

    @GetMapping("/hello")
    fun hello(@RequestParam(defaultValue = "world") name: String) = Greeting("Hello, " + name + "!")
}
